state/controller.go

```diff
@@ -25,6 +29,46 @@ func controllerKey(controllerUUID string) string {
 	return fmt.Sprintf("%s#%s", controllerGlobalKey, controllerUUID)
 }
 
+// Controller encapsulates state for the Juju controller as a whole,
+// as opposed to model specific functionality.
+type Controller struct {
+	clock                  clock.Clock
+	controllerModelTag     names.ModelTag
+	controllerTag          names.ControllerTag
+	mongoInfo              *mongo.MongoInfo
+	session                *mgo.Session
+	policy                 Policy
+	newPolicy              NewPolicyFunc
+	runTransactionObserver RunTransactionObserverFunc
+}
+
+// Close the connection to the database.
+func (ctlr *Controller) Close() error {
+	return errors.Trace(ctlr.session.Close())
+}
+
+// NewState returns a new State instance for the specified model. The
+// connection uses the same credentials and policy as the Controller.
+func (ctlr *Controller) NewState(modelTag names.ModelTag) (*State, error) {
+	session := ctlr.session.Copy()
+	st, err := newState(
+		modelTag,
+		ctlr.controllerModelTag,
+		session,
+		ctlr.mongoInfo,
+		ctlr.newPolicy,
+		ctlr.clock,
+		ctlr.runTransactionObserver,
+	)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	if err := st.start(ctlr.controllerTag); err != nil {
```

Comment:
From a "does this api look obviously correct" standpoint, it seems odd to pass in newState(...controllerModelTag)
but then have to pass st.start(controllerTag0.
If the State object needs a pointer to the controller separate from the controller model, I would still expect to pass that information into "newState" not have to pass it into start().
I have the feeling this is just cruft from when we decided that controllers should be a thing separate from the model that is that controller. (though IIRC, the actually share a UUID anyway).